resource/resource.go

```diff
@@ -158,6 +160,8 @@ func (r *Resource) Find(ctx context.Context, lookup *Lookup, page, perPage int)
 	if r.storage == nil {
 		return nil, ErrNoStorage
 	}
+	r.lockResources.Lock()
+	defer r.lockResources.Unlock()
```

Comment:
Are you sure this mutex is needed? I don't see how parallel `Find` would generate a race.